Graphics/ShaderTools/src/ShaderToolsCommon.cpp

```diff
@@ -157,12 +157,38 @@ void AppendShaderSourceCode(std::string& Source, const ShaderCreateInfo& ShaderC
     Source.append(SourceData.Source, SourceData.SourceLength);
 }
 
+static void GePositionInfo(const Char* pBuffer, size_t BufferSize, size_t Offset, size_t& Line, size_t& LineOffset)
+{
+    VERIFY_EXPR(Offset < BufferSize);
+
+    LineOffset = Offset;
+    for (size_t i = 0; i < Offset; ++i)
+    {
+        if (pBuffer[i] == '\n')
+        {
+            ++Line;
+            LineOffset = Offset - i;
+        }
+    }
+}
+
+static String ParserErrorMessage(const char* Message, const Char* pBuffer, size_t BufferSize, size_t Offset)
+{
+    size_t Line       = 0;
+    size_t LineOffset = 0;
+    GePositionInfo(pBuffer, BufferSize, Offset, Line, LineOffset);
+
+    std::stringstream Stream;
+    Stream << Message << "[" << Line << ":" << LineOffset << "]";
+    return Stream.str();
+}
+
 // https://github.com/tomtom-international/cpp-dependencies/blob/a91f330e97c6b9e4e9ecd81f43c4a40e044d4bbc/src/Input.cpp
-template <typename HandlerType>
-void FindIncludes(const char* pBuffer, size_t BufferSize, HandlerType IncludeHandler)
+template <typename HandlerType, typename ErrorHandlerType>
+bool FindIncludes(const char* pBuffer, size_t BufferSize, HandlerType IncludeHandler, ErrorHandlerType ErrorHandler)
```

Comment:
The function becomes much easier to read if you use current position pointer instead of the offset